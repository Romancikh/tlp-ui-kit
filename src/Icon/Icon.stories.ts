import type { Meta, StoryObj } from "@storybook/react";

import Icon from "./Icon";

const meta = {
  title: "Icon",
  component: Icon,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  argTypes: {
    color: { control: "select" },
  },
} satisfies Meta<typeof Icon>;

export default meta;
type Story = StoryObj<typeof meta>;

export const AddIcon: Story = {
  args: {
    name: "AddIcon",
  },
};

export const Colored: Story = {
  args: {
    name: "AddIcon",
    color: "success",
  },
};

export const AutoStoriesIcon: Story = {
  args: {
    name: "AutoStoriesIcon",
  },
};

export const CancelOutlinedIcon: Story = {
  args: {
    name: "CancelOutlinedIcon",
  },
};

export const CheckCircleOutlineIcon: Story = {
  args: {
    name: "CheckCircleOutlineIcon",
  },
};

export const DeleteOutlineIcon: Story = {
  args: {
    name: "DeleteOutlineIcon",
  },
};

export const EditIcon: Story = {
  args: {
    name: "EditIcon",
  },
};

export const MenuIcon: Story = {
  args: {
    name: "MenuIcon",
  },
};

export const NavigateBeforeIcon: Story = {
  args: {
    name: "NavigateBeforeIcon",
  },
};
