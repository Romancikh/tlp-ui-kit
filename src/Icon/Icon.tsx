import AddIcon from "@mui/icons-material/Add";
import AutoStoriesIcon from "@mui/icons-material/AutoStories";
import CancelOutlinedIcon from "@mui/icons-material/CancelOutlined";
import CheckCircleOutlineIcon from "@mui/icons-material/CheckCircleOutline";
import DeleteOutlineIcon from "@mui/icons-material/DeleteOutline";
import EditIcon from "@mui/icons-material/Edit";
import MenuIcon from "@mui/icons-material/Menu";
import NavigateBeforeIcon from "@mui/icons-material/NavigateBefore";

const icons = {
  AddIcon,
  AutoStoriesIcon,
  CancelOutlinedIcon,
  CheckCircleOutlineIcon,
  DeleteOutlineIcon,
  EditIcon,
  MenuIcon,
  NavigateBeforeIcon,
};

type IconProps = {
  name: keyof typeof icons;
  color?: "success" | "primary" | "error" | "secondary";
};

function Icon({ name, color }: IconProps) {
  const SelectedIcon = icons[name];
  return <SelectedIcon color={color} />;
}

export default Icon;
