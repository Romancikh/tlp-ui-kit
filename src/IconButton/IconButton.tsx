import MuiIconButton from "@mui/material/IconButton";
import { PropsWithChildren } from "react";

type IconButtonProps = PropsWithChildren & {
  onClick?: () => void;
};

function IconButton({ onClick, children }: IconButtonProps) {
  return <MuiIconButton onClick={onClick}>{children}</MuiIconButton>;
}

export default IconButton;
