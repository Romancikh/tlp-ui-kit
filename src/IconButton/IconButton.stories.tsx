import type { Meta, StoryObj } from "@storybook/react";

import Icon from "../Icon/Icon";
import IconButton from "./IconButton";

const meta = {
  component: IconButton,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  title: "IconButton",
} satisfies Meta<typeof IconButton>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: ({ ...args }) => (
    <IconButton {...args}>
      <Icon name="AddIcon" />
    </IconButton>
  ),
};
