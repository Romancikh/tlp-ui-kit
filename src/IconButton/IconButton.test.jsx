import { render } from "@testing-library/react";
import Icon from "../Icon/Icon";
import IconButton from "./IconButton";

test("IconButton renders correctly", () => {
  const { container } = render(
    <IconButton>
      <Icon name="AddIcon" />
    </IconButton>,
  );
  expect(container).toMatchSnapshot();
});
