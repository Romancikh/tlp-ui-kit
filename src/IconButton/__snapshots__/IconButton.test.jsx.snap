// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders correctly 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="AddIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
